sbi/analysis/plot.py

```diff
@@ -1396,12 +1700,9 @@ def _arrange_grid(
                 if excl_diag:
                     ax.axis("off")  # pyright: ignore reportOptionalMemberAccess
                 else:
-                    for sample_ind, sample in enumerate(samples):
-                        diag_f = diag_funcs[sample_ind]
-                        if callable(diag_f):  # is callable:
-                            diag_f(
-                                ax, sample[:, row], limits[row], diag_kwargs[sample_ind]
-                            )
+                    diag_f = diag_funcs[0]
```

Comment:
please add a TODO here pointing to the issue with the unsupported list of `diag_kwargs`